lib/wasi/src/runtime/task_manager/mod.rs

```diff
@@ -70,6 +83,21 @@ pub trait VirtualTaskManager: std::fmt::Debug + Send + Sync + 'static {
         spawn_type: SpawnType,
     ) -> Result<(), WasiThreadError>;
 
+    /// Starts an WebAssembly task will will run on a dedicated thread
+    /// pulled from the worker pool that has a stateful thread local variable
+    /// After the trigger has successfully completed
+    fn resume_wasm_after_trigger(
+        &self,
+        task: Box<dyn FnOnce(Store, Module) + Send + 'static>,
+        store: Store,
+        module: Module,
+        #[allow(clippy::all)] trigger: Box<
```

Comment:
As the clippy lint says ... let's move that out to a separate type definition `type WhateverTrigger = Box<...>`.

Reply:
fixed